container.go

```diff
@@ -244,18 +244,14 @@ func fetchContainer(pod *Pod, containerID string) (*Container, error) {
 		return nil, errNeedContainerID
 	}
 
-	fs := filesystem{}
-	config, err := fs.fetchContainerConfig(pod.id, containerID)
-	if err != nil {
-		return nil, err
-	}
-
-	container, err := createContainer(pod, config)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create container with config %v in pod %v: %v", config, pod.id, err)
+	for _, c := range pod.containers {
+		if containerID == c.id {
+			return c, nil
+		}
```

Comment:
A couple of things on that one:

* This needs a comment.
* Having to do so means we're fetching a container from storage while having a fully built pod with all its container already fetched and added. When is that happening ?